Source/BoundaryConditions/PML_RZ.cpp

```diff
@@ -0,0 +1,252 @@
+/* Copyright 2019 Andrew Myers, Aurore Blelly, Axel Huebl
+ * Maxence Thevenet, Remi Lehe, Weiqun Zhang
+ *
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#include "PML_RZ.H"
+
+#include "BoundaryConditions/PML_RZ.H"
+#include "BoundaryConditions/PMLComponent.H"
+#ifdef WARPX_USE_PSATD
+#   include "FieldSolver/SpectralSolver/SpectralFieldDataRZ.H"
+#endif
+#include "Utils/WarpXAlgorithmSelection.H"
+#include "Utils/WarpXConst.H"
+#include "Utils/WarpXProfilerWrapper.H"
+#include "WarpX.H"
+
+#include <AMReX.H>
+#include <AMReX_Algorithm.H>
+#include <AMReX_Array.H>
+#include <AMReX_Array4.H>
+#include <AMReX_BLassert.H>
+#include <AMReX_Box.H>
+#include <AMReX_BoxList.H>
+#include <AMReX_DistributionMapping.H>
+#include <AMReX_FArrayBox.H>
+#include <AMReX_FBI.H>
+#include <AMReX_FabArrayBase.H>
+#include <AMReX_Geometry.H>
+#include <AMReX_GpuControl.H>
+#include <AMReX_GpuDevice.H>
+#include <AMReX_GpuLaunch.H>
+#include <AMReX_GpuQualifiers.H>
+#include <AMReX_IndexType.H>
+#include <AMReX_MFIter.H>
+#include <AMReX_ParmParse.H>
+#include <AMReX_RealVect.H>
+#include <AMReX_SPACE.H>
+#include <AMReX_VisMF.H>
+
+#include <algorithm>
+#include <cmath>
+#include <memory>
+#include <utility>
+
+using namespace amrex;
+
+PML_RZ::PML_RZ (const int lev, const BoxArray& grid_ba, const DistributionMapping& grid_dm,
+                const Geometry* geom, const int ncell, const int do_pml_in_domain)
+    : m_ncell(ncell),
+      m_do_pml_in_domain(do_pml_in_domain),
+      m_geom(geom)
+{
+
+    m_ok = true;
+
+    auto & Er_fp = WarpX::GetInstance().getEfield_fp(lev,0);
+    auto & Et_fp = WarpX::GetInstance().getEfield_fp(lev,1);
+    pml_E_fp[0] = std::make_unique<MultiFab>( amrex::convert( grid_ba, Er_fp.ixType().toIntVect() ), grid_dm,
+                                              Er_fp.nComp(), Er_fp.nGrow() );
+    pml_E_fp[1] = std::make_unique<MultiFab>( amrex::convert( grid_ba, Et_fp.ixType().toIntVect() ), grid_dm,
+                                              Et_fp.nComp(), Et_fp.nGrow() );
+
+    auto & Br_fp = WarpX::GetInstance().getBfield_fp(lev,0);
+    auto & Bt_fp = WarpX::GetInstance().getBfield_fp(lev,1);
```

Comment:
I would be in favor of using the explicit types as much as possible, if you agree:
```suggestion
    amrex::MultiFab & Br_fp = WarpX::GetInstance().getBfield_fp(lev,0);
    amrex::MultiFab & Bt_fp = WarpX::GetInstance().getBfield_fp(lev,1);
```

Reply:
Fixed